src/herder/HerderSCPDriver.cpp

```diff
@@ -324,39 +342,29 @@ HerderSCPDriver::validateValueHelper(uint64_t slotIndex, StellarValue const& b,
         return SCPDriver::kInvalidValue;
     }
 
-    if ((!nomination || lcl.ledgerVersion < 11) &&
-        b.ext.v() != STELLAR_VALUE_BASIC)
-    {
-        // ballot protocol or
-        // pre version 11 only supports BASIC
-        CLOG(TRACE, "Herder")
-            << "HerderSCPDriver::validateValue"
-            << " i: " << slotIndex << " invalid value type - expected BASIC";
-        return SCPDriver::kInvalidValue;
-    }
-    if (nomination &&
-        (lcl.ledgerVersion >= 11 && b.ext.v() != STELLAR_VALUE_SIGNED))
-    {
-        // v11 and above use SIGNED for nomination
-        CLOG(TRACE, "Herder")
-            << "HerderSCPDriver::validateValue"
-            << " i: " << slotIndex << " invalid value type - expected SIGNED";
-        return SCPDriver::kInvalidValue;
-    }
-
     Hash const& txSetHash = b.txSetHash;
     TxSetFramePtr txSet = mPendingEnvelopes.getTxSet(txSetHash);
 
     SCPDriver::ValidationLevel res;
 
+    // In the presence of the "closetime change" for protocol issue 622,
+    // the close time to validate against is not the last ledger close
+    // time, but the proposed close time (which we have validated above,
+    // in checkCloseTime(), is greater than the last ledger close time).
+    if (b.closeTime <= lastCloseTime)
```

Comment:
That's why I called it a `logic_error`; I wanted to make sure it'd be detected if something changed earlier in the method so that that wasn't guaranteed anymore.  I'll remove it.